### PR TITLE
Add AttributeStore & related data components

### PR DESCRIPTION
**Pull request by mpriour**
Tests pull requests will be issued after pull request  #35 is dealt with as they depend on the restructuring included in that request

### CODE REVIEW

**Comment by mpriour** [comment]
Ignore comment above regarding pull requests #35 and the tests.
Tests are in a84146c and all pass

File: src/GXM/data/AttributeModel.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+/**
+ * @class GXM.data.AttributeModel
+ * <p>A specific model for WFS DescribeFeatureType records.<p>
+ *
+ * Preconfigured with an Ajax proxy and a
+ * {@link GXM.data.reader.Attribute}.
+ */
+
```
Comment (bartvde) [inline src/GXM/data/AttributeModel.js:20]: The rest of GXM does not use JSDoc as yet for api docs. I agree this is the way forward though.

File: src/GXM/data/OwsStore.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ * See https://github.com/geoext/geoext2/blob/master/license.txt for the full text
+ * of the license.
+ */
+
+/**
+ * <p>Small Base class to make creating stores for remote OWS information sources
+ * easier.</p>
+ * <p>NOTE: This is a BASE CLASS and is not designed for direct use in an
+ * application. Instead, one should extend from this class in any situation in
+ * which a you need a {@link Ext.data.proxy.Server} (ex: 'ajax', 'jsonp', etc) and a
```
Comment (bartvde) [inline src/GXM/data/OwsStore.js]: should be "which you need" ?

File: src/GXM/data/OwsStore.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+ * reader which requires an {@link OpenLayers.Format} to parse the data.</p>
+ *
+ * @class GeoExt.data.OwsStore
```
Comment (bartvde) [inline src/GXM/data/OwsStore.js]: GeoExt -> GXM

File: src/GXM/data/OwsStore.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ */
+Ext.define('GXM.data.OwsStore', {
+    extend: 'Ext.data.Store',
+    alternateClassName: ['GXM.data.OWSStore'],
+
+    config: {
+        /**
+         * @cfg {String}
+         * The URL from which to retrieve the WMS DescribeLayer document
+         */
```
Comment (bartvde) [inline src/GXM/data/OwsStore.js:27]: The comment should not reference WMS DescribeLayer document, I am guessing this is copy/paste.

File: src/GXM/data/OwsStore.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        url : null,
+
+        /**
+         * @cfg {OpenLayers.Format}
+         * A parser for transforming the XHR response into an array of objects
+         * representing attributes. Defaults to an {OpenLayers.Format.WMSDescribeLayer}
+         * parser.
+         */
```
Comment (bartvde) [inline src/GXM/data/OwsStore.js:34]: Comment is also not accurate here.

File: src/GXM/data/reader/Attribute.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+        ignore: null
+    },
+    
+    /**
+     * Create a new reader.
+     * @param {Object} config (optional) Config object.
+     */
+    constructor: function(config) {
+        if (!this.model) {
+            this.model = 'GeoExt.data.AttributeModel';
```
Comment (bartvde) [inline src/GXM/data/reader/Attribute.js:64]: GeoExt -> GXM

File: src/GXM/data/reader/Attribute.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+     */
+    ignoreAttribute: function(name, value) {
+        var ignore = false,
+            ignoreConfig = this.getIgnore();
+        if(ignoreConfig && ignoreConfig[name]) {
+            var matches = ignoreConfig[name];
+            if(typeof matches == "string") {
+                ignore = (matches === value);
+            } else if(matches instanceof Array) {
+                ignore = (matches.indexOf(value) > -1);
```
Comment (bartvde) [inline src/GXM/data/reader/Attribute.js]: safer to use Ext.Array.indexOf

**Comment by bartvde** [comment]
@mpriour this looks good, I've put some minor comments in the code, if you can fix those then please merge.

File: src/GXM/data/AttributeModel.js (hunk 1 above)
Comment (mpriour) [inline src/GXM/data/AttributeModel.js:20]: The rest of GXM was converted to JSDoc as of 3406c8d9e

Comment (bartvde) [inline src/GXM/data/AttributeModel.js:20]: Ah I had missed that. Thanks.

**Comment by bartvde** [comment]
@mpriour any chance you can update this one so that it can be merged? TIA.

**Comment by mpriour** [comment]
Yes but I won't have a chance to look at it until tonight at the earliest

**Comment by bartvde** [comment]
Tonight will be fine @mpriour TIA

**Comment by mpriour** [comment]
@bartvde That last commit should allow everything to be merged without conflict

**Comment by bartvde** [comment]
@mpriour the diff of the pull request still has changes that should not be in there

**Comment by bartvde** [comment]
I'm having issues cleaning up this PR in the current issue number, so I am gonna create a new one in favour of this one

**Pull request closed without merging**